utils/istio/istio.go

```diff
@@ -64,19 +64,42 @@ func GetVirtualServiceNamespaceName(vsv string) (string, string) {
 	return namespace, name
 }
 
+func MultipleVirtualServiceConfigured(rollout *v1alpha1.Rollout) bool {
+	if rollout.Spec.Strategy.Canary.TrafficRouting.Istio.VirtualServices != nil {
+		return true
+	}
+	return false
+}
+
 // GetRolloutVirtualServiceKeys gets the referenced VirtualService and its namespace from a Rollout
 func GetRolloutVirtualServiceKeys(ro *v1alpha1.Rollout) []string {
+	var virtualServices []v1alpha1.IstioVirtualService
+	var virtualServiceKeys []string
 	canary := ro.Spec.Strategy.Canary
-	if canary == nil || canary.TrafficRouting == nil || canary.TrafficRouting.Istio == nil || canary.TrafficRouting.Istio.VirtualService.Name == "" {
+
+	if canary == nil || canary.TrafficRouting == nil || canary.TrafficRouting.Istio == nil ||
+		(canary.TrafficRouting.Istio.VirtualServices == nil && canary.TrafficRouting.Istio.VirtualService.Name == "") {
 		return []string{}
 	}
 
-	namespace, name := GetVirtualServiceNamespaceName(canary.TrafficRouting.Istio.VirtualService.Name)
-	if namespace == "" {
-		namespace = ro.Namespace
+	if MultipleVirtualServiceConfigured(ro) {
+		if canary.TrafficRouting.Istio.VirtualService.Name != "" {
```

Comment:
this should not happen, right? Because the validation logic will not allow to reach this place. If so, this is a panic.
Or we can remove line 86 - 88

Reply:
Such scenario should be detected earlier in the validationVirtualServiceConfig, I believe this if and return empty string array may introduce a subtle bug in the future.

Reply:
please resolve this comment

Reply:
Yes, this check is already taken care. I have removed it.
Please check, let me know if you are okay with this change then I will resolve it.